pkg/codegen/schema.go

```diff
@@ -500,6 +535,9 @@ func GenStructFromSchema(schema Schema) string {
 		objectParts = append(objectParts,
 			fmt.Sprintf("AdditionalProperties map[string]%s `json:\"-\"`", addPropsType))
 	}
+	if len(schema.UnionElements) != 0 {
+		objectParts = append(objectParts, "union json.RawMessage")
```

Comment:
Thanks a lot for your PR, we use it for quite a while already!
Recently we encountered an issue with it thought, so hopefully you are willing to update the PR, so it won't be an issue when it merged merged anymore:

When using `anyOf` as a type in combination with `additionalProperties` like this:
```
variables:
  type: object
  description: Extra variables of type bool, number, or string
  additionalProperties:
    oneOf:
      - type: boolean
      - type: number
      - type: string
```
([spec](https://swagger.io/specification/#schema-object) states that for `additionalObject` "... inline or referenced schema MUST be of a Schema Object", so this seems to be allowed)

it is not possible to get the actual value, because of the private `union`.

For example the `Get()` function is generated as follows:
```
func (a Variables) Get(fieldName string) (value struct {
	union json.RawMessage
}, found bool) {
	if a.AdditionalProperties != nil {
		value, found = a.AdditionalProperties[fieldName]
	}
	return
}
```
Which returns a `struct` with the private `union` field. Since it comes from a different package, the field is not accessible.

A similar scenario can be imagined when using the `Set()` function (it requires the struct with the private field, which is not accepted as the same type by the compiler)

Also, using `for-range` to iterate over the `Variables.AdditionalProperties` is not possible due to the private field.

Changing `union` to `Union` (and fixing the template (`union.tmpl`) to use `Union`) fixes this issue for me.

Reply:
Thanks, I've added a test case and forced generator to create a new type for additional properties, if this type has union elements or additional properties of its own. Also found and fixed merging union cases with allOf

Reply:
Thanks a lot, your addition works nicely for our case :+1: